rocky/tools/forms/boefje.py

```diff
@@ -65,3 +63,66 @@ class BoefjeAddForm(BaseRockyForm):
         widget=forms.Select(choices=SCAN_LEVEL.choices),
         help_text=BOEFJE_SCAN_LEVEL_HELP_TEXT,
     )
+
+    def __init__(self, katalogus_client: KATalogusClientV1, plugin_id: str, created: str, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.katalogus_client = katalogus_client
+        self.plugin_id = plugin_id
+        self.created = created
+
+    def create_boefje_with_form_data(self, form_data, plugin_id: str, created: str):
+        arguments = [] if form_data["oci_arguments"] == "" else form_data["oci_arguments"].split()
+        consumes = [] if form_data["consumes"] == "" else form_data["consumes"].strip("[]").replace("'", "").split(", ")
+        produces = [] if form_data["produces"] == "" else form_data["produces"].split(",")
+        produces = [p.strip() for p in produces]
+        input_objects = []
+
+        for input_object in consumes:
+            input_objects.append(type_by_name(input_object))
+
+        return Boefje(
+            id=plugin_id,
+            name=form_data["name"],
+            created=created,
+            description=form_data["description"],
+            enabled=False,
+            type="boefje",
+            scan_level=form_data["scan_level"],
+            consumes=input_objects,
+            produces=produces,
+            schema=form_data["schema"],
+            oci_image=form_data["oci_image"],
+            oci_arguments=arguments,
+        )
+
+
+class BoefjeAddForm(BoefjeSetupForm):
+    def clean(self):
+        cleaned_data = super().clean()
+
+        plugin = self.create_boefje_with_form_data(cleaned_data, self.plugin_id, self.created)
+
+        try:
+            self.katalogus_client.create_plugin(plugin)
+        except ReadTimeout:
+            handle_existing_name(self, plugin.name)
```

Comment:
I dont think creating the plugin should be done / tried in the clean method.

Reply:
Normally in a ModelForm of Django they do it in the form in the `save` method. After form validation for example save this user. The Django normal Forms does not have a save method, but the clean method instead. It is the same as doing it in the views in the `form_valid` method. I think we can create a boefje in the clean, otherwise in the view it would go in the `form_valid` which is basically the same thing. Doing it in the form has some advantages as we can throw a form error before the form is posted. In the view this is not possible then to create a django message, which we need to do redirects.

Reply:
Agree with @underdarknl

`clean` and `clean_<field>` should be used to validate (and sanitize) the form. Creation of resources should be preferably done *after* form validation, and in a `form_valid` method (meaning the view should inherit from a form view or `FormMixin`

Now this is a bit of a different situation, since we want field validation and the only method here to validate seems to be creating the plugin and expect an error. But I still think this should be handled differently. For example, instead of creating the plugin, we should check for existence using another way (e.g. an endpoint that you can reach to see wether the plugin exists, this could be the endpoint used by the client's `get_plugin` method). So in short: in the clean/ validation use a get plugin call to check for existence, and in the form valid you should submit/ create the plugin